packages/gatsby-plugin-gatsby-cloud/src/components/Indicator.js

```diff
@@ -27,64 +252,41 @@ export default function Indicator({ children }) {
 
     if (!buildId) {
       if (isOnPrettyUrl) {
-        buildId = latestBuild?.id
+        setBuildId(latestBuild?.id)
       } else {
         // Match UUID from preview build URL https://build-af44185e-b8e5-11eb-8529-0242ac130003.gtsb.io
         const buildIdMatch = host.match(/build-(.*?(?=\.))/)
-        buildId = buildIdMatch && buildIdMatch[1]
+        if (buildIdMatch) {
+          setBuildId(buildIdMatch[1])
+        }
       }
     }
 
-    const defaultBuildInfo = {
-      createdAt: currentBuild?.createdAt,
-      orgId: siteInfo?.orgId,
-      siteId: siteInfo?.siteId,
-      buildId,
+    setBuildInfo({
+      currentBuild,
+      latestBuild,
+      siteInfo,
       isOnPrettyUrl,
-      sitePrefix: siteInfo?.sitePrefix,
+    })
+
+    if (shouldPoll.current) {
+      setTimeout(pollData, POLLING_INTERVAL)
     }
```

Comment:
We should stoe this timeout in a variable and clear it when we unmount. Else your page might crash when the component was unmounted and a timeout got kicked of.
```suggestion
    let pollRef
    if (shouldPoll.current) {
      pollRef = setTimeout(pollData, POLLING_INTERVAL)
    }
    
    return () => {
      if (pollRef) {
        clearTimeout(pollRef);
      }
    }
```

Reply:
Good point!